### ISSUE TITLE
Rule proposal: Consistent decorator position for exports

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#2038]
### Description

Decorators can be placed in two positions when you export a class. Would be nice with a rule to enforce either for consistency.

https://twitter.com/leobalter/status/1621242353494425602

I haven't yet made up my mind which should be the default. Lets discuss.

Either:

```js
export default @decorator class {}
```

or

```js
@decorator export default class {}
```

My mind is split as I kinda feel that export should always come first, since it should immediately clear what's being exported. On the other hand, placing it first will allow stacking it above (the rule may want to enforce whether this is allowed too):

```js
@decorator
export default class {}
```

I'm leaning towards the latter.

### Fail

TODO

### Pass

TODO

**Comment by fregante** [comment]
I'd place it above especially because decorators are stackable and callable too, so:

``` js
@foo
@bar(opts)
export default class Baz {}
``` 

The alternatives don't seem great once you have multiple decorators and real names


``` js
export default @foo @bar(opts) class Baz{}
``` 

Or even

``` js 
export default 
@foo 
@bar(opts) 
class Baz{ 😱 }
```

**Comment by fisker** [comment]
Isn't decorator before "export" legacy? The proposal should already updated. If you are talking about decorators in typescript, then ignore my comment, it may support it forever, who knows.

**Comment by fisker** [comment]
> Class decorators come after `export` and `default`.

https://github.com/tc39/proposal-decorators#syntax

**Comment by fisker** [comment]
```js
export default 
@foo 
@bar(opts) 
class Baz{ 😱 }
```

You can do

```js
@foo 
@bar(opts) 
class Baz { 😱 }

export default Baz
```

Because it's a declaration.

**Comment by sindresorhus** [comment]
@fisker See the linked tweet. The decision was just recently made to support both styles.

**Comment by fisker** [comment]
Sorry, didn't see that.

I'd say before `export` should be preferred.

**Comment by sindresorhus** [comment]
> I'd say before export should be preferred then, but not on same line with export.

That's what I'm leaning towards too.

**Comment by sindresorhus** [comment]
Seems there needs to be 3 styles to choose from:

- `above`
  ```js
  @decorator
  export default class {}
  ```
- `before`
  ```js
  @decorator export default class {}
  ```
- `after`
  ```js
  export default @decorator class {}
  ```

**Comment by fisker** [comment]
It seems there is already a plugin https://github.com/NullVoxPopuli/eslint-plugin-decorator-position

**Comment by sindresorhus** [comment]
@fisker I think it still makes sense to add here:

1. The rule does not actually control whether the decorator should be before or after, which is the main goal of this rule.
2. People (including me) don't want to add a plugin just for one rule.
3. The rule there has too many options.

**Comment by sindresorhus** [comment]
Accepted
